subsys/net/lib/dns/mdns_responder.c

```diff
@@ -46,6 +46,8 @@ static struct net_context *ipv6;
 #endif
 
 static struct net_mgmt_event_callback mgmt_cb;
+static struct dns_sd_rec *external_records;
```

Comment:
Unless the external records are modified in this compilation unit (which I assume is not the intent), this should probably be `const` (i.e. the data pointing to should not be modified).
```suggestion
static const struct dns_sd_rec *external_records;
```